rime_main.c

```diff
@@ -85,23 +170,27 @@ void ibus_rime_start(gboolean full_check) {
   ibus_rime_traits.distribution_code_name = DISTRIBUTION_CODE_NAME;
   ibus_rime_traits.distribution_version = DISTRIBUTION_VERSION;
   ibus_rime_traits.app_name = "ibus";
-  static RIME_MODULE_LIST(ibus_rime_modules, "default", "legacy");
-  ibus_rime_traits.modules = ibus_rime_modules;
+  ibus_rime_traits.modules = NULL;
+
+  // first initialization
   rime_api->initialize(&ibus_rime_traits);
   if (rime_api->start_maintenance((Bool)full_check)) {
     // update frontend config
     rime_api->deploy_config_file("ibus_rime.yaml", "config_version");
   }
-}
 
-static void* legacy_module_handle = NULL;
-
-static void load_plugin_modules() {
-  legacy_module_handle = dlopen("librime-legacy.so", RTLD_LAZY);
+  // parse & load plugin modules
+  if (!load_plugins("ibus_rime")) {
+    rime_api->finalize();
```

Comment:
Yes. it is safe to do so.
Re-initializing without calling `finalize()` should work unless module loading order is important because of side effects:
without `finalize()`, loading an existing module is no-op, this may change loading order of modules.

For example: (This is totally theoretical. I doubt if anyone would ever hit a bug like this.)
default modules: `a, b`;
plugin `C` introduces new module `c`, and the defined order is `a, c, b` in `ibus_rime.yaml`, but actual order they are loaded is `a, b, (loading_plugins) c`.
This wouldn't be a problem unless `c` registers a component `user_db` which is already provided by `b`.
In actual, the `c` version of `user_db` replaces the `b` version, which the YAML author intended to avoid by having `b` register later.